File: blankToAsterisk/JavaScript/blankToAsterisk.js
/*
 * Question:
 *  Write a program to enter a line of input and replace
 *  all blanks with asterisks '*'
 *
 * Sample run:
 *  Please enter a line:
 *  This is a text  with normal   blanks.
 *
 *  Line with asterisks:
 *  This*is*a*text***with*normal***asterisks.
 * */

function main(){
    // Strings
    let sUsrInput, sResult;

    // Normally, I could have written my own character replacement
    // function by iterating the string searching for blank character.
    // But since the reason of learning javascript is about my web project
    // which is related to my company, being quick on learning js is more
    // important than writing an algorithm which I have already done it
    // with C programming language. So I chose to use 'replaceAll()' function
    // to get it done.
    sUsrInput = window.prompt('Please enter a line:\n');
    sResult = sUsrInput.replaceAll(" ","*");

    console.log("Line with asterisks:\n" + sResult);
}

main();

File: blankToAsterisk/C++/blankToAsterisk.cpp
/*
 * Question:
 *  Write a program to enter a line of input and replace
 *  all blanks with asterisks '*'
 *
 * Sample run:
 *  Please enter a line:
 *  This is a text  with normal   blanks.
 *
 *  Line with asterisks:
 *  This*is*a*text***with*normal***asterisks.
 * */


#include <iostream>
#include <stdio.h>
#include <string.h>
using namespace std;
int main(void){
    char s_input[256];
    char s_input2[256];
    int i=0;

    // Input.
    cout << "Enter a string: ";
    //cin >> s_input; (For some reason, cin does not accept blank spaces.
    //                 So I used fgets to take the input)
    fgets(s_input,256,stdin); // fgets append an aditional '\n' to the end of the string.


    // Output
    cout << "Before parsing: " << s_input << endl;
    for(i=0; i<(strlen(s_input)-1); i++){
        // Why it wants me to use single quotes instead of double for comparison ??
        // Answer: Single quotes are defined to express character literals.
        if(s_input[i] == ' '){
            s_input[i] = '*';
        }
    }
    cout << "After parsing: " << s_input << endl;
    return 0;
}